frontend/src/components/elements/Video/Video.tsx

```diff
@@ -42,7 +42,40 @@ class Video extends React.PureComponent<Props> {
   }
 
   public render(): React.ReactNode {
+    /* Element may contain "url" or "data" property. */
+
     const { element, width } = this.props
+
+    if (element.get("url")) {
+      /* is this a YouTube link? if so we need a fancier tag. 
+         NOTE: This part assumes the URL is already an "embed" link.
+      */
+      if (element.get("url").includes("youtube.com")) {
```

Comment:
Can we use a stronger YouTube pattern-matcher?

And, actually, I think the cleanest solution here would be to have the Python side do this work, and then add a field to the proto telling the frontend to use an iframe.

Something like
```

message Video {
  oneof src {
    // The data in Base64 format.
    string data = 1;

    // A url pointing to a video file
    string url = 4;
  }

  enum Type {
    UNUSED = 0;  // This should always exist.
    NATIVE = 1;
    IFRAME = 2;
  }

  Type type = 5;

  // The type attribute of the HTML <video> tag's <source> subtag.
  // ex: "video/mp4"
  string format = 2;

  // The currentTime attribute of the HTML <video> tag's <source> subtag.
  int32 start_time = 3;
}
```

Reply:
OK -- per convo in the office:

* YouTube iframes have enough special stuff going on that we should make a YouTube-iframe type in the protobuf.
* Rename generic_binary_proto (file) to media_proto (refactor both video and audio in DeltaGenerator)
* media_proto.marshall should handle and pack urls and do special things to video URLs. 
* Move YouTube url processor into media_proto as private function

Video Protobuf changes:

```
message Video {
  oneof src {
    // The data in Base64 format.
    string data = 1;

    // A url pointing to a video file
    string url = 4;
  }

  enum Type {
    UNUSED = 0;  // This should always exist.
    NATIVE = 1;
    YOUTUBE_IFRAME = 2;
  }

  Type type = 5;

  // The type attribute of the HTML <video> tag's <source> subtag.
  // ex: "video/mp4"
  string format = 2;

  // The currentTime attribute of the HTML <video> tag's <source> subtag.
  int32 start_time = 3;
}
```

Reply:
These changes are now in!